core/pkg/server/sender.go

```diff
@@ -1134,6 +1136,54 @@ func (s *Sender) sendSync(record *service.Record, request *service.SyncRequest)
 	s.fwdChan <- rec
 }
 
+func (s *Sender) sendStopStatus(record *service.Record, _ *service.StopStatusRequest) {
+	// TODO: ideally we could get it direclty from the settings
+	entity := s.RunRecord.GetEntity()
+	project := s.RunRecord.GetProject()
+	runId := s.RunRecord.GetRunId()
+
+	result := &service.Result{
+		ResultType: &service.Result_Response{
+			Response: &service.Response{
+				ResponseType: &service.Response_StopStatusResponse{
+					StopStatusResponse: &service.StopStatusResponse{
+						RunShouldStop: false,
+					},
+				},
+			},
+		},
+		Control: record.Control,
+		Uuid:    record.Uuid,
+	}
```

Comment:
A helper function would be cleaner here rather than mutating this value

```go
func makeResult(shouldStop bool) *service.Result {
	...
}

...

if err != nil {
	...
	s.outChan <- makeResult(/*shouldStop*/ false)
	return
}

...
s.outChan <- makeResult(/*shouldStop*/ ...)
```

Reply:
i wanted to do it in a refactor, maybe i should have case handling... for now and do your thing is a general makeResult logic